modules/core/src/main/java/org/apache/ignite/internal/processors/query/QuerySysIndexDescriptorImpl.java

```diff
@@ -59,11 +58,11 @@ public QuerySysIndexDescriptorImpl(String name, Collection<String> fields) {
 
     /** {@inheritDoc} */
     @Override public QueryIndexType type() {
-        throw new NotImplementedException();
+        throw new RuntimeException("Not implemented");
```

Comment:
What about to use `UnsupportedOperationException` here and below?

Reply:
Sounds reasonable. +1